### DISCUSSION TITLE
SupervisedTemplate giving errors

### DISCUSSION THREAD

**Discussion by h-0-0** [ContinualAI/avalanche#1436] [Q&A] [answered]
Hello, I'm trying to use the SupervisedTemplate, however, it's not working and I'm getting the following errors:

> File "/home/jd18380/Documents/FixedStream/run.py", line 85, in regular
    cl_strategy = SupervisedTemplate(
  
> File`` "/home/jd18380/Documents/venv_CL-exp/lib/python3.8/site-packages/avalanche/training/templates/common_templates.py", line 128, in __init__super().__init__()  # type: ignore

> File "/home/jd18380/Documents/venv_CL-exp/lib/python3.8/site-packages/avalanche/training/templates/observation_type/batch_observation.py", line 18, in __init__super().__init__()

> TypeError: __init__() missing 2 required positional arguments: 'model' and 'optimizer'

I'm calling SupervisedTemplate as follows:
`    cl_strategy = SupervisedTemplate(
        model, optimizer,
        criterion = CrossEntropyLoss(), train_epochs=epochs,
        train_mb_size = batch_size, eval_mb_size = batch_size,
        evaluator=eval_plugin,
        device = device
    )`
So am unsure as to why I am getting an error, as I am providing both the model and the optimizer.

**Comment by AntonioCarta** [comment, marked as answer]
This is due to a backward-incompatible change in `typing_extensions`. We are still investigating this problem, but you can temporarily solve it by using `typing_extensions==4.4.0`.

**Reply by h-0-0** [reply]
Thanks! this fixed it.
